h2/src/main/org/h2/mvstore/db/ValueDataType.java

```diff
@@ -592,6 +605,17 @@ private Object readValue(ByteBuffer buff) {
         }
         case SPATIAL_KEY_2D:
             return getSpatialDataType().read(buff);
+        case CUSTOM_DATA_TYPE: {
+            int customType = readVarInt(buff);
+            int len = readVarInt(buff);
+            byte[] b = DataUtils.newBytes(len);
+            buff.get(b, 0, len);
+            Value v = ValueBytes.getNoCopy(b);
+            if (JdbcUtils.customDataTypesHandler != null) {
+                return JdbcUtils.customDataTypesHandler.convert(v, customType);
+            }
+            return v;
```

Comment:
Looks like we have to throw an exception if we can not convert value instead of returning ValueBytes.